JavaScript/JavaScriptSDK/Util.ts

```diff
@@ -220,8 +220,14 @@ module Microsoft.ApplicationInsights {
         /**
          * helper method to set userId and sessionId cookie
          */
-        public static setCookie(name, value) {
-            Util.document.cookie = name + "=" + value + ";path=/";
+        public static setCookie(name, value, domain?) {
+            var domainAttrib = "";
+
+            if (domain) {
+                domainAttrib = ";domain=" + domain;
```

Comment:
Yes, user can use an inline function to dynamically set the cookie domain. Sample code: 

 ```
           instrumentationKey: "<your-key>",
            cookieDomain: (function () { return 1 === 2 ? "check your math!" : "we're good" })()
```